### PR TITLE
Graphics clear funtions

### PR DESCRIPTION
**Pull request by chaoticbob**
This CL adds missing clear functions for render targets and depth stencils that clear images within a render pass.

- Added clear functions for render targets and depth stencil.
- Added depth/stencil clear flag enum bits.
- Added functions to expose mip level and array layer in RTV and DSV.
- Added logic to determine if grfx::RenderPass contains any attachments with clear ops.
- Changed clear count check in grfx::CommandBuffer::BeingRenderPass to only check if there's a clear attachment.
- Added function to grfx::RenderPass to return index of an image.
- Added clear function implementation to Vulkan and D3D12.
- Modified 04_cube to use clear functions.

### CODE REVIEW

File: include/ppx/grfx/grfx_command.h [hunk 1]
```diff
@@ -245,13 +245,27 @@ class CommandBuffer
     CommandBuffer() {}
     virtual ~CommandBuffer() {}
 
+    grfx::CommandType GetCommandType() { return mCreateInfo.pPool->GetCommandType(); }
```
Comment (wangra-google) [inline include/ppx/grfx/grfx_command.h:248]: const?

File: projects/04_cube/Cube.cpp [hunk 2]
```diff
@@ -220,19 +244,22 @@ void CubeApp::Render()
...
-        beginInfo.DSVClearValue             = {1.0f, 0xFF};
+
+        // Clear RTV to greyish blue
+        grfx::RenderTargetClearValue rtvClearValue = {0.23f, 0.23f, 0.33f, 0};
+        grfx::DepthStencilClearValue dsvClearValue = {1.0f, 0xFF};
 
         frame.cmd->TransitionImageLayout(renderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_RENDER_TARGET);
         frame.cmd->BeginRenderPass(&beginInfo);
         {
+            frame.cmd->ClearRenderTarget(renderPass->GetRenderTargetImage(0), rtvClearValue);
```
Comment (wangra-google) [inline projects/04_cube/Cube.cpp:261]: why do we move to explicit clear instead of doing the clear in the beginrenderpass?

Comment (chaoticbob) [reply projects/04_cube/Cube.cpp:261]: The commit message says that this change to the sample was to test the implementation of the clear functions.

File: include/ppx/grfx/grfx_command.h (hunk 1 above)
Comment (chaoticbob) [reply include/ppx/grfx/grfx_command.h]: Yes. Fixed.

**Review by wangra-google — APPROVED** [review]

**Pull request merged**